src/serial/RS232Device.cc

```diff
@@ -22,24 +22,36 @@ void RS232Device::setParityBit(bool /*enable*/, ParityBit /*parity*/)
 	// ignore
 }
 
+// The input lines of the RS-232 port without any plugged in device
+// are left floating (bits read as 1 in the IO ports), with the exception
+// of the following devices:
+//
+// Spectravideo SVI-738 : Inputs are pull-up (Read as 0 in the IO ports)
+// Spectravideo SVI-737 and SVI-757: Inputs are pull-down, behave 
+//                                   the same as left floating.
+//
+// This has been tested to be true for the SVI-738 and Toshiba HX-22,
+// for the other devices the info comes from manuals, schematics, or
+// circuit board pictures.
+
 bool RS232Device::getCTS(EmuTime::param /*time*/) const
 {
-	return true; // TODO check
+	return false; // TODO return the inverse of config tag 'rs232_pullup'
```

Comment:
Thanks a lot for your very detailed investigation.
In openMSX we don't often take pull-up, pull-down (or really 'floating') into account. But you're absolutely right that we should.

I have a suggestion for the implementation:
* You added a `<rs232_pullup>` tag in the MSX RS232 device. IMHO this is correct
* In the commit message you talk about passing a 'DeviceConfig` object to RS232Device. IMHO that's not correct.

My reasoning is the following:
* Suppose you have a hypothetical MSX machine with two RS232 interfaces, one with pull-up resistors, and the other with pull-down.
* You have a RS232 peripheral with some unconnected pins.
* If you plug this peripheral in one interface the corresponding status bit will read as '1', in the other interface it will read as '0'.
* So the same (unchanging) peripheral behaves different depending on in which interface it gets plugged. That's an indication that the `<rs232_pullup>` tag indeed belongs in the interface.

But still, from a practical point of view, in a method like `bool getCTS()` we must either return `true` or `false`. And which is the correct value depends on whether pull-up/down resistors are present. And I assume that's why you suggested to pass this information (via a `DeviceConfig` object) to `RS232Device`.

I would propose a different approach:
* A real pin in a RS232 connector can be in 3 states: it can be driven high, driven low, or it can be unconnected (=floating).
* But in the current implementation we model the value of a pin with a `bool`, which can only have 2 states.

Therefor to resolve this I would propose to not return a `bool` but something else which _can_ have 3 states. The first (simplest?) thing that comes to mind is a `std::optional<bool>`:
* An empty optional then represents an unconnected pin.
* An optional with a value represents a driven high/low pin.

--

I 100% agree with you that we should implement the difference between pull-up/pull-down. And I'd like to encourage you to continue working on this. But I propose to do that in a separate PR.

So for this PR that `TODO` comment is fine.

Reply:
> I would propose a different approach:
> 
> * A real pin in a RS232 connector can be in 3 states: it can be driven high, driven low, or it can be unconnected (=floating).
> * But in the current implementation we model the value of a pin with a `bool`, which can only have 2 states.
> 
> Therefor to resolve this I would propose to not return a `bool` but something else which _can_ have 3 states. The first (simplest?) thing that comes to mind is a `std::optional<bool>`:
> 
> * An empty optional then represents an unconnected pin.
> * An optional with a value represents a driven high/low pin.
> 

Yes I think this would be a simple and effective way to implement it.
My first idea was to implement it in RS232device the way I described in the commit, then I figured to implement it in MSXRS232 would be better but couldn't find a way to distinguish when something was plugged in. 

> --
> 
> I 100% agree with you that we should implement the difference between pull-up/pull-down. And I'd like to encourage you to continue working on this. But I propose to do that in a separate PR.
> 
> So for this PR that `TODO` comment is fine.